tests/pipe/action/test_retry.py

```
@@ -0,0 +1,178 @@
+    assert [action.id for action in result.actions] == ["B1"]
+    assert pipe.forward_incremental.await_count == 1
+
+
+@pytest.mark.asyncio
+async def test_forward_raises_parsed_context_size_error_without_retrying() -> None:
+    error = RuntimeError(
+        "Please reduce the length of the messages or completions. Current length is 200 while limit is 100"
+    )
+    wrapper, pipe = make_wrapper(forward_side_effect=error, max_tries=3)
+
+    with pytest.raises(ContextSizeTooLargeError, match="200.*100"):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)